### ISSUE TITLE
Terraform plan is getting crashed with the error: "panic: not a collection type"

### ISSUE THREAD

**Issue by harshagv** [terraform-providers/terraform-provider-azurerm#8379]
Terraform plan is getting crashed with the error: "panic: not a collection type"

**terraform versions**: 
```
- random v2.3.0
- template v2.1.2
- azurerm v2.26.0
- terraform v0.13.1
```

Attached is crash and console logs

**terraform Configuration Files:**
main.tf
```
provider "azurerm" {
  version = "~>2.0"
  features {}
}

data "azurerm_resource_group" "azresgroup" {
  name = var.az_res_group
}

data "azurerm_subnet" "azsubnet" {
  name = var.az_subnet
  virtual_network_name = var.az_virtual_nw_name
  resource_group_name  = var.az_res_group
}

data "azurerm_network_security_group" "nsg_name" {
  name = var.az_nsg_name
  resource_group_name = var.az_res_group
}

data "azurerm_public_ip" "simplified_stack_public_ip" {
  name = var.az_public_ip_name
  resource_group_name = var.az_res_group
}

data "template_file" "vm_init_script" {
  template = file("../scripts/initialize_setup.sh")
}

resource "random_id" "randomId" {
  byte_length = 8
}

resource "azurerm_network_interface" "nw_interface_name" {
  name = "var.az_nw_interface_name${random_id.randomId.hex}"
  location = var.az_location
  resource_group_name = var.az_res_group
  ip_configuration {
    name = "var.az_ip_configuration${random_id.randomId.hex}"
    subnet_id = data.azurerm_subnet.azsubnet.id
    private_ip_address_allocation = "static"
    private_ip_address = "10.0.2.5"
    public_ip_address_id = data.azurerm_public_ip.simplified_stack_public_ip.id
  }

  tags = {
    owner: "gvh"
  }
}

resource "azurerm_network_interface_security_group_association" "nw_interface_nsg_assoc" {
  network_interface_id = azurerm_network_interface.nw_interface_name.id
  network_security_group_id = data.azurerm_network_security_group.nsg_name.id
}

resource "azurerm_managed_disk" "simplified_stack_mgd_disk" {
  name = "var.az_managed_disk_name${random_id.randomId.hex}"
  location = var.az_location
  resource_group_name = var.az_res_group
  storage_account_type = var.az_storage_acc_type
  create_option = "Empty"
  disk_size_gb = var.az_managed_disk_size

  tags = {
    environment = "staging-db"
  }
}

resource "azurerm_virtual_machine" "simplified_stack_vm" {
  name = var.az_vm_name
  location = var.az_location
  resource_group_name = var.az_res_group
  network_interface_ids = [ azurerm_network_interface.nw_interface_name.id ]
  vm_size = var.az_vm_size

  storage_image_reference {
    publisher = "Canonical"
    offer = "UbuntuServer"
    sku = "18.04-LTS"
    version = "latest"
  }

  storage_os_disk {
    name = "az_os_disk_name${random_id.randomId.hex}"
    create_option = "FromImage"
    caching = "ReadWrite"
    disk_size_gb = var.az_os_disk_size
  }

  os_profile {
    computer_name = var.az_vm_name
    admin_username = var.az_vm_admin_user_name
    custom_data = base64encode(data.template_file.vm_init_script.rendered)
  }

  os_profile_linux_config {
    disable_password_authentication = true
    ssh_keys {
      path = "/home/${var.az_vm_admin_user_name}/.ssh/authorized_keys"
      key_data = file("../scripts/ssh-key.pub")
    }
  }

  tags = {
    environment = var.env_type,
    owner: "gvh"
  }
}

resource "azurerm_virtual_machine_data_disk_attachment" "attach_managed_disk" {
  managed_disk_id = azurerm_managed_disk.simplified_stack_mgd_disk.id
  virtual_machine_id = azurerm_virtual_machine.simplified_stack_vm.id
  lun = "10"
  caching = "ReadWrite"
}
```

**Attachments:**
[crash.log](https://github.com/terraform-providers/terraform-provider-azurerm/files/5184021/crash.log)
[console_logs.txt](https://github.com/terraform-providers/terraform-provider-azurerm/files/5184024/console_logs.txt)

**Steps to Reproduce**
```
- az login
- az account set -s <id>
- terraform init
- terraform plan
```

**Comment by njuCZ** [comment]
Hi @harshagv, this seems a regression bug for terraform 0.13.0 and 0.13.1. you could refer to this issue https://github.com/hashicorp/terraform/issues/26011. the latest release terraform version 0.13.2 has fixed this issue. Could you please upgrade your terraform and have a try again?

**Comment by harshagv** [comment]
@njuCZ Yes, issue got fixed with using terraform version `0.13.2`. Thanks for your comments.
